clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -1217,6 +1217,8 @@ static void handlePreferredName(Sema &S, Decl *D, const ParsedAttr &AL) {
 
 bool Sema::isValidPointerAttrType(QualType T, bool RefOkay) {
   if (RefOkay) {
+    if (T->isDependentType())
```

Comment:
This doesn't seem like the right place for this, we should be skipping ANY time the type is dependent, not only when a ref argument is acceptable.

Reply:
Think you, updated